util/plain_hasher/src/lib.rs

```diff
@@ -17,13 +17,14 @@
 #[macro_use]
 extern crate crunchy;
 extern crate ethereum_types;
+extern crate hashdb;
 
 use ethereum_types::H256;
-use std::collections::{HashMap, HashSet};
+use hashdb::Hasher;
 use std::hash;
-
+use std::collections::{HashMap, HashSet};
 /// Specialized version of `HashMap` with H256 keys and fast hashing function.
-pub type H256FastMap<T> = HashMap<H256, T, hash::BuildHasherDefault<PlainHasher>>;
+pub type H256FastMap<H, T> = HashMap<<H as Hasher>::Out, T, hash::BuildHasherDefault<PlainHasher>>;
```

Comment:
This is invalid. `plain_hasher` has an assumption that key is 32 bytes long hash. It will not work for hashes of different length

Reply:
Indeed there's a REVIEW question for figuring out what to do about this [here](https://github.com/paritytech/parity/pull/8739/files#diff-2109d58f1e6b9a9260dc72aa1cd58979R184).
I know Polkadot wants to use `Blake2b` and will probably be mostly 32 bytes. Lifting the 32 byte assumption seems like it requires some care but perhaps we can tackle that in a separate PR where we figure out a way to make the length configurable (conditional compilation? macro? generics?)?

Reply:
can this type be defined as associated type of Hasher that you introduced? 

```rust
trait Hasher {
  type Out: AsRef<u8>;
  type StdHasher: std::hash::Hasher;
}
```

Reply:
Hmm. Not sure I understand. You mean something like this?

```rust
pub type HashedFastMap<H, T> = HashMap<<H as Hasher>::Out, T, hash::BuildHasherDefault<H::StdHasher>>;
```

It feels like which (hash map) hasher to use is orthogonal to the key construction algo.

Would it be hard/impossible to refactor `PlainHasher` to be generic over the length? e.g. `PlainHasher<L>` and then pass in `KeccakHasher::LENGTH`?